### PR TITLE
Upgrade compile and target sdk version to Android API 33

### PR DESCRIPTION
**Pull request by fluiddot**
## Description

This PR is related to the upgrade effort to Android 13 in Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/issues/5556. It basically updates all references of `compileSdkVersion` and `targetSdkVersion` parameters to use version `33`.

This change is similar to when we [upgrade to Android 12](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/23).

## Test plan

This update can be tested using the Gutenberg editor and the following PR: https://github.com/WordPress/gutenberg/pull/50731.
The idea would be to run smoke testing on the demo project in order to check that there are no issues with Android 13.

### CODE REVIEW

File: android/lib/src/main/java/com/swmansion/gesturehandler/PanGestureHandler.kt
Comment (fluiddot) [inline android/lib/src/main/java/com/swmansion/gesturehandler/PanGestureHandler.kt:1]: This change has been copied from the upstream repository: https://github.com/software-mansion/react-native-gesture-handler/commit/8d70e87b035e6c5d2009d41f4137a10a1b77ca5a

**Review by geriux — APPROVED** [review]

**Comment by fluiddot** [comment]
@geriux Note that this PR is merged into `wp-fork-2.3.2` as the previous Android update was also made this way. I noticed that `wp-fork` uses version `2.5.0`, however, I haven't found that it has ever been used in Gutenberg 🤔. Since you created [this PR](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/22) (probably related to an RN upgrade), do you know which version we should use? Thanks 🙇 !

**Comment by geriux** [comment]
Oh yes! We had an issue on iOS where the editor would freeze so we decided to keep an older version, but the fork kept `2.5.0` so we should keep using `2.3.2` until we figure out those issues, possibly already fixed in a newer version of the library.

**Comment by fluiddot** [comment]
> Oh yes! We had an issue on iOS where the editor would freeze so we decided to keep an older version, but the fork kept `2.5.0` so we should keep using `2.3.2` until we figure out those issues, possibly already fixed in a newer version of the library

Ah, thanks for sharing the info. I'll add a note about this in the [`wp-fork` PR](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/2) as a warning for future changes.

**Pull request merged**